bluefog/common/nccl_controller.cc

```diff
@@ -577,9 +579,17 @@ void NCCLController::NeighborAllgather(TensorTableEntry& entry) {
     displcmnts = new int[mpi_ctx_.neighbor_indgree_];
     status = mpi_ctx_.AllocateOutput(entry, recvcounts, Communicator::GRAPH);
   } else {
-    bool is_topo_check_fail = CheckNeighborSendRecvPattern(
-        mpi_ctx_.size_, entry, timeline_ptr_,
-        mpi_ctx_.GetMPICommunicator(Communicator::GLOBAL));
+    bool is_topo_check_fail = false;
+    if (entry.enable_topo_check && entry.dynamic_neighbors_enabled) {
+      if (entry.is_hierarchical) {
+        // TODO: support check.
+        BFLOG(INFO) << "Request to check topology for hierarchical neighbor "
+                    << "allreduce ops but it is not supported yet.";
+      }
```

Comment:
should be  if-else structure? Same comments for other places
```
if (entry.is_hierarchical) {
  warning...
} else {
  CheckNeighborSendRecvPattern(..)
}
```

Reply:
Done.